app/models/follow_up.rb

```diff
@@ -19,19 +19,22 @@ def send_to_api
   private
 
   def body
-    "subscriber[route_id]=#{destination.route_id}&subscriber[language_code]=#{language.code}"\
-    "&subscriber[email]=#{email}#{names}#{auth_params}"
+    auth_params.merge(subscriber: subscriber_params).to_query
   end
 
-  def names
-    return nil if name.nil?
+  def auth_params
+    { access_id: destination.access_key_id, access_secret: destination.access_key_secret }
+  end
 
-    names = name.split(' ')
-    "&subscriber[first_name]=#{names[0]}&subscriber[last_name]=#{names[1]}"
+  def subscriber_params
+    { route_id: destination.route_id, language_code: language.code, email: email }.merge(name_params)
```

Comment:
do I need to do anything special to deal with the fact that `name_params` might return `nil`?

Reply:
I'm not sure.  I'd say check the documentation on `merge.`